### ISSUE TITLE
Question about pseudo label and category mismatch. 

### ISSUE THREAD

**Issue by ShenZheng2000** [CityU-AIM-Group/SIGMA#43]
**Pseudo Label**
From what I gather, the graphs for target images in the paper are constructed solely based on the pseudo label due to the lack of ground truth labels. 

However, in cases where the domain gap between source and target is very large, such as going from a sunny day to a heavy rainy night, relying on the pseudo label can be inadequate and may lead to issues in graph construction.

Assuming my understanding is accurate, are there any potential solutions to address this problem and enhance SIGMA's performance?

**Category Mismatch**
I have an additional concern regarding the effectiveness of the node completion (DNC) strategy used in the paper. The datasets used for domain adaptation, such as Cityscape, FoggyCityscape, Sim10k, and KITTI, have similar categories. 

As a result, I am uncertain whether DNC would perform well if the categories were significantly different between the source and target datasets.

**Comment by wymanCV** [comment]
Hi, thanks for your good questions. I am happy that you have such deep and insightful thoughts and show interest in our work.

Pseudo Label
The smoothness property of the graph operations will make it robust for noisy labels. Therefore, we set a very low threshold (0.5) to sample nodes. More insights can be found in several noisy-label-related works, e.g.,  the famous graph propagation in pseudo-labeling (semi-supervised learning) [here](https://arxiv.org/abs/1904.04717). This is a theoretically-grounded idea, which hasn't been considered in DAOD.

To further improve the performance, you can consider other scenarios related to noisy labels to find the solution. E.g., introduce Teacher-Student EMA, add CycleGAN interpolated data (Note that this will significantly improve the self-training-based method), and logit engineering.

Category Mismatch
The main function of DNC is to enable the unbiased adaptation signal in each iteration, designed for the cross-domain inconsistent P(Y). Yes, we also found that it will make a more noticeable effect in the scenarios with more classes, e.g., P2C. Moreover, this can be seen as a fine-grained version of the existing prototype-based approach (RPN Prototype Alignment for Domain Adaptive Object Detector).

Hope this can help you a lot.

**Comment by ShenZheng2000** [comment]
Thanks for your timely reply. Your response is really helpful to me!
